src/python/pants/rules/core/cloc.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pathlib import Path
+from typing import Set
+
+from pants.base.build_environment import get_buildroot
+from pants.base.specs import Specs
+from pants.engine.console import Console
+from pants.engine.fs import (Digest, DirectoriesToMerge, FilesContent, InputFileContent, Snapshot,
+                             UrlToFetch)
+from pants.engine.goal import Goal, LineOriented
+from pants.engine.isolated_process import ExecuteProcessRequest, ExecuteProcessResult
+from pants.engine.legacy.graph import HydratedTargets, TransitiveHydratedTargets
+from pants.engine.rules import console_rule, rule
+from pants.engine.selectors import Get
+from pants.util.objects import datatype
+
+
+class DownloadedClocScript(datatype([('script_path', str), ('digest', Digest)])):
+  """Cloc script as downloaded from the pantsbuild binaries repo."""
+
+
+@rule(DownloadedClocScript, [])
+def download_cloc_script():
```

Comment:
By hardcoding this, folks won't be able to change the version any longer.

I know that this isn't the first place that we've done this, but would you mind creating/referencing a ticket about adding support to the existing Subsystems for specifying per-platform digests for things, so that our existing options (--version, --binary-base-url, etc) can be used?